### DISCUSSION TITLE
Incorrect version number for rure

### DISCUSSION THREAD

**Discussion by sternenseemann** [Homebrew/discussions#3919] [Tap maintenance and Homebrew development]
Hi, I'm the maintainer for `rure` in nixpkgs. From [the repology page](https://repology.org/project/rure/versions) I've noticed that you are following the version number of the `regex` crate for `rure` as well.

`rure` has an independent version number (see <https://crates.io/crates/rure>, <https://github.com/rust-lang/regex/blob/master/regex-capi/Cargo.toml#L3>) which I believe would be more correct to use. Since the `regex` version is always higher, it kind of messes with the up to date state in repology which is a bit annoying.

It'd be great if homebrew could also use the `rure` version number, so this becomes a bit more consistent across package repositories.  Unfortunately I have no clue how homebrew determines the version number from the formula, so I haven't proposed this change myself.

cc @rolandcrosby, @carlocab (original authors), @chenrui333 (frequent contributor to the formula)

**Comment by Bo98** [comment]
Thanks for letting us know! From first glance, I agree that the wrong version number was picked here.

But looking at it further it seems complicated. `rure` has this dependency: https://github.com/rust-lang/regex/blob/f871a8eb1d725a8aaa56e2ceea57c24fce74b5fd/regex-capi/Cargo.toml#L22.

When pulling through `cargo` it will get the latest `regex` available but when building from a tarball/git clone then it'll take the regex included in that download.

`rure` (`regex-capi`) gets updated very rarely (3 year gaps between each of the last couple versions), while `regex` is updated a lot more regularly. So it's not super clear the best way to keep up-to-date with `regex` changes while keeping the `rure` versioning. Simply using the `regex` version was probably not correct though.

**Reply by rolandcrosby** [reply]
[Here's the original discussion](https://github.com/Homebrew/homebrew-core/pull/94091#discussion_r797212263) of Homebrew versioning for this library. As mentioned above, the problem is that the rure/regex-capi version is not directly related to the version of the regex crate. The most correct way to handle this might be to include both version numbers somehow, so any new version of either the C bindings or the regex crate implementation would be treated as a new version of the whole library. But I don't know whether that's something Homebrew can support, or how to get that to play nicely with the versioning scheme of the regex crate's GitHub releases.

**Reply by sternenseemann** [reply]
`rure` also uses semver, so it claims that when the version number is not changed it doesn't matter with what regex version it is built in practice as long as it satisfies the Cargo.toml. Since it also doesn't ship a Cargo.lock, you are free to choose any regex version that works.

In nixpkgs we just generate a lock file to figure out which dependencies to fetch (since we built from the crates.io tarball of rure, not the GitHub repository for regex).

I think you could still keep the regex tarball up to date and use the rure version number, as it is internal to the package in a sense – maybe encode it via the formula version (if you have such a concept)?
